### ISSUE TITLE
Update all state machines to avoid self transitions and hidden logic

### ISSUE THREAD

**Issue by mfateev** [temporalio/sdk-java#193]
Some workflow state machines have self transitions and additional conditional logic which is not part of the modeled state machine. For example, ActivityStateMachine has  these self transitions:

```
SCHEDULE_COMMAND_CREATED --> SCHEDULE_COMMAND_CREATED: SCHEDULE_ACTIVITY_TASK
STARTED_ACTIVITY_CANCEL_COMMAND_CREATED --> STARTED_ACTIVITY_CANCEL_COMMAND_CREATED: REQUEST_CANCEL_ACTIVITY_TASK
SCHEDULED_ACTIVITY_CANCEL_COMMAND_CREATED --> SCHEDULED_ACTIVITY_CANCEL_COMMAND_CREATED: REQUEST_CANCEL_ACTIVITY_TASK
```
These should be eliminated by introducing additional states as each of the events that cause state transition can be applied only once.

Activity `cancellationType` is modeled using if/else conditions in ActivityStateMachine. This should be converted to additional states in the state machine itself.

The goal of these changes is to make sure that all of the complex logic related to activity execution is explicitly modeled. This would simplify the understanding of the logic and help with the creation of additional SDKs.
